### ISSUE TITLE
Setting own profile to RTL scrambles others' non-RTL posts view

### ISSUE THREAD

**Issue by mabkenar** [tootsuite/mastodon#2588]
This is very related to #2350 but slightly different: it concerns direction settings for the **skeleton** of the page. This issue is introduced in release 1.3.1

### What I see
If I set my profile setting to Persian (an RTL language), then this is how I see a normal [LTR post](https://social.tchncs.de/@jaltek/1409432):

![selection_182](https://cloud.githubusercontent.com/assets/3006332/25534773/3564a9e0-2c35-11e7-8c34-365450bb0767.png)
[looking messy]

### What others see
Another user with profile set to English/German/etc (or me with a Private browser window that does not know about my profile preferences) sees the page like this:

![selection_183](https://cloud.githubusercontent.com/assets/3006332/25534764/2f68276a-2c35-11e7-94c1-d519a9043911.png)
[looking fine]

Problems with the first view:

- user avatar of the main post is shown at left, while username is at right.
- user avatar for the reply post is shown at right and is blocking the time stamp ("an hour ago") of the reply post

### What should happen
Display of posts should not depend of my profile preferences. It should depend on the **post** language. For RTL posts, the avatar and username should appear at right, and the time stamp should appear at left. Moreover, margins should be set accordingly so that the avatar does not cover any text.

As long as this issue exists ~~the human race will never be free~~ setting the profile to an RTL language will breaks **rendering of all non-RTL posts**.

* * * *

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This bug happens on a [tagged release](https://github.com/tootsuite/mastodon/releases) and not on `master` (If you're a user, don't worry about this).

**Comment by mabkenar** [comment]
I would really like to solve (or help to solve) this issue because this is breaking my usage of Mastodon. Based on a [recent commit](https://github.com/tootsuite/mastodon/pull/3682/files) by @Gargron, I see that these two files are related to this bug:

https://github.com/tootsuite/mastodon/blob/master/app/views/stream_entries/_detailed_status.html.haml
https://github.com/tootsuite/mastodon/blob/master/app/views/stream_entries/_simple_status.html.haml

And this is what each of the two files render:
![3564a9e0-2c35-11e7-8c34-365450bb0767](https://user-images.githubusercontent.com/3006332/27168302-5e4a6b0c-51a5-11e7-939e-529086a88276.png)

I.e., the first one renders a single toot (or the prominent toot in a conversation) and the second one renders other toots listed on the same page, or all the toots in the user's main profile view.

The problem is, I can't yet see where the styles are set. If I want to fix how the skeleton of the page is built, I probably need to change the stylesheet. Any help or ideas?

**Comment by mabkenar** [comment]
For another example of scrambled layouts, see [this profile page](https://persadon.com/@masoud) of mine (image below). You see that boost icons are on left, while boost action descriptions are at right. Moreover, profile picture covers timestamp and part of the toot text.

I see my profile like this because I have set my interface language to Persian (فارسی).

![selection_404](https://user-images.githubusercontent.com/3006332/29363015-7ffa980c-828e-11e7-899f-58603e5e798a.png)

**Comment by mabkenar** [comment]
The most severe problems mentioned in this issue were fixed with [this commit](https://github.com/tootsuite/mastodon/commit/26402ee2cb57b4c0695c584fa24acfe429c14547) (Thank you, Gargron :-) ). Profile pictures no longer cover timestamps and toots. Also, boost icons are shown next to booster's name, where they should be.
However, there is still the (less severe) problem: When I set **my** interface to an RTL language, **all other profiles on the local server** are displayed right-justified, although content from those users are not RTL. For example, this is how I see [another profile page on glitch.social](https://glitch.social/@bea):

![selection_432](https://user-images.githubusercontent.com/3006332/30250676-fd20aa92-9652-11e7-94d9-7ecaf66cc477.png)

I close this issue for now and follow the progress on the remaining problems in separate issues.
